src/sql/hydra.sql

```diff
@@ -85,8 +85,8 @@ create table Jobsets (
     type          integer not null default 0, -- 0 == legacy, 1 == flake
     flake         text,
     check (schedulingShares > 0),
-    check ((type = 0) = (nixExprInput is not null and nixExprPath is not null)),
-    check ((type = 1) = (flake is not null)),
+    constraint jobsets_check_legacy check ((type != 0) OR (nixExprInput is not null and nixExprPath is not null and flake is     null)),
```

Comment:
Now that we have identified that this is already too complex for the "average developer" to maintain we might want to add a comment here to inform the next person coming along.

The commit message is doing a pretty good job on explaining how it is supposed to be. Mind adding two lines here that do the same?